extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -96,20 +121,44 @@ export class ScrollManager {
 
   /**
    * @param {!../../../src/service/viewport/viewport-interface.ViewportChangedEventDef} e
+   * @return {!Promise}
    * @private
    */
   onScroll_(e) {
-    /** {./scroll-manager.ScrollEventDef} */
-    const scrollEvent = {
-      top: e.top,
-      left: e.left,
-      width: e.width,
-      height: e.height,
-      scrollWidth: this.viewport_.getScrollWidth(),
-      scrollHeight: this.viewport_.getScrollHeight(),
-    };
-    // Fire all of our children scroll observables
-    this.scrollObservable_.fire(scrollEvent);
+    const firstMeasure = this.initialRootLayoutRect_ === null;
+    return this.getInitRootElementRect_().then(initialRootLayoutRect => {
+      const {
+        height: initialScrollHeight,
+        width: initialScrollWidth,
+      } = initialRootLayoutRect;
+
+      return (firstMeasure
```

Comment:
Since you're already keeping track of whether this is the `firstMeasure`, this ternary is redundant.
Use just `this.getInitRootElementRect_()` instead.